### PR TITLE
test: add integration tests for borrow document

### LINKED ISSUE #196: BorrowDocument tests

### PR DESCRIPTION
**Pull request by StarryFolf**
fix: time

### CODE REVIEW

File: tests/Application.Tests.Integration/BaseClassFixture.cs [hunk 1]
```diff
@@ -207,8 +207,8 @@ protected static Borrow CreateBorrowRequest(User borrower, Document document, Bo
             Document = document,
             Reason = "something something",
             Status = status,
-            BorrowTime = LocalDateTime.FromDateTime(DateTime.UtcNow),
-            DueTime = LocalDateTime.FromDateTime(DateTime.UtcNow + TimeSpan.FromDays(1))
+            BorrowTime = LocalDateTime.FromDateTime(DateTime.Now),
+            DueTime = LocalDateTime.FromDateTime(DateTime.Now + TimeSpan.FromDays(1))
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/BaseClassFixture.cs:211]: I really don't think using DateTime.Now is a good practice, but we'll investigate later

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyNotFoundException_WhenUserDoesNotExist()
+    {
+        // Arrange
+        var command = new BorrowDocument.Command()
+        {
+            BorrowerId = Guid.NewGuid(),
+            DocumentId = Guid.NewGuid(),
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs]: Document should exist to prove the point of borrower does not exist

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    {
+        // Arrange
+        var user = CreateUser(IdentityData.Roles.Employee, "aaaaaa");
+        user.IsActivated = false;
+        await AddAsync(user);
+        
+        var command = new BorrowDocument.Command()
+        {
+            BorrowerId = user.Id,
+            DocumentId = Guid.NewGuid(),
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs]: document should exist

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+            .WithMessage("This document is already requested borrow from the same user.");
+        
+        // Cleanup;
+        Remove(borrow);
+        Remove(user);
+        Remove(document);
+        Remove(department);
+    }
+
+    [Fact] public async Task ShouldThrowConflictException_WhenARequestIsMadeWhileDocumentIsAlreadyBeingBorrowed()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs]: what is this positioning

**Review by ChienNQuang — COMMENTED** [review]
Just questioning

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+        
+        // Cleanup
+        Remove(user);
+    }
+    
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenUserIsNotActivated()
+    {
+        // Arrange
+        var document = CreateNDocuments(1).First();
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:133]: I think you forgot to cleanup documents

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("User does not exist.");
+    }
+    
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenUserIsNotActive()
+    {
+        // Arrange
+        var document = CreateNDocuments(1).First();
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:100]: here too

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+        Remove(user);
+        Remove(document);
+        Remove(await FindAsync<Department>(department.Id));
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyNotFoundException_WhenUserDoesNotExist()
+    {
+        // Arrange
+        var document = CreateNDocuments(1).First();
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:72]: here too

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
a little bit more and it should be fine

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs (hunk 5 above)
Comment (StarryFolf) [reply tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:133]: done

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs (hunk 6 above)
Comment (StarryFolf) [reply tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:100]: done

File: tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs (hunk 7 above)
Comment (StarryFolf) [reply tests/Application.Tests.Integration/Borrows/Commands/BorrowDocumentTests.cs:72]: done

**Review by ChienNQuang — APPROVED** [review]
looks good to me

**Pull request merged**